hazelcast/src/test/java/com/hazelcast/cp/internal/WipeDestroyedObjectsTest.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.cp.internal;
+
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.cp.CPGroupId;
+import com.hazelcast.cp.CPSubsystem;
+import com.hazelcast.cp.IAtomicLong;
+import com.hazelcast.cp.IAtomicReference;
+import com.hazelcast.cp.ICountDownLatch;
+import com.hazelcast.cp.ISemaphore;
+import com.hazelcast.cp.internal.datastructures.atomiclong.AtomicLongService;
+import com.hazelcast.cp.internal.datastructures.atomicref.AtomicRefService;
+import com.hazelcast.cp.internal.datastructures.countdownlatch.CountDownLatchService;
+import com.hazelcast.cp.internal.datastructures.lock.LockService;
+import com.hazelcast.cp.internal.datastructures.semaphore.SemaphoreService;
+import com.hazelcast.cp.internal.datastructures.spi.atomic.RaftAtomicValueService;
+import com.hazelcast.cp.internal.datastructures.spi.atomic.RaftAtomicValueServiceAccessor;
+import com.hazelcast.cp.internal.datastructures.spi.blocking.AbstractBlockingService;
+import com.hazelcast.cp.internal.datastructures.spi.blocking.ResourceRegistry;
+import com.hazelcast.cp.internal.datastructures.spi.blocking.ResourceRegistryAccessor;
+import com.hazelcast.cp.lock.FencedLock;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Collection;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Consumer;
+
+import static com.hazelcast.test.Accessors.getNodeEngineImpl;
+import static org.junit.Assert.assertTrue;
+
+public class WipeDestroyedObjectsTest extends HazelcastRaftTestSupport {
+    public static final Set<String> SERVICES = Set.of(LockService.SERVICE_NAME, SemaphoreService.SERVICE_NAME,
+            CountDownLatchService.SERVICE_NAME, AtomicRefService.SERVICE_NAME, AtomicLongService.SERVICE_NAME);
+    public static final Set<String> ATOMIC_SERVICES = Set.of(AtomicLongService.SERVICE_NAME, AtomicRefService.SERVICE_NAME);
+    private static final Set<String> CP_GROUP_NAMES = Set.of("mygroup1", "mygroup2", "default");
+    private HazelcastInstance[] instances;
+
+    private CPSubsystem getCpSubsystem() {
+        return instances[0].getCPSubsystem();
+    }
+
+    @Before
+    public void setup() {
+        instances = newInstances(3);
+    }
+
+    @Test
```

Comment:
Personally, I'd be happy to have some simple tests to check if we can recreate an object with the same name, which is the main purpose of this PR.
```
Lock iLock = cpSubsystem.getLock("iLock);
AtomicLong iLong =  cpSubsystem.getAtomicLong("iLong);
...
iLock.IsLocked();
iLong.get();
...
cpSubsystemManagementService.wipeDestroyedObjects();

Lock iLock = cpSubsystem.getLock("iLock);
AtomicLong iLong =  cpSubsystem.getAtomicLong("iLong);
...
iLock.IsLocked();
iLong.get();
...
```

Reply:
The tests had that logic added in addition to what was there previously. Happy to simplify to move things forward if that's holding things up.

Reply:
Simplified.